google_directory_service.py

```diff
@@ -72,3 +73,21 @@ def GetUsersByGroupKey(self, group_key):
 
     return users
 
+  def SearchForUser(self, user_key):
+    """Search for a user based on a user key.
+
+    Args:
+      user_key: A string identifying an individual user.
+
+    Returns:
+      users: A list with that user in it or empty.
+      List format is used here for consistency with the other methods.
+    """
+    users = []
+    request = self.service.users().get(userKey=user_key, projection='full')
+    result = request.execute(num_retries=NUM_RETRIES)
+    if result['primaryEmail']:
+      users.append(result)
+
+    return users
```

Comment:
I don't think so, no. The API says if successful it will return 'a Users resource' which just means the user and associated fields according to the projection we request (https://developers.google.com/admin-sdk/directory/v1/reference/users/get#response). However, I'm returning it as a list for uniformity with the other methods.